python/smqtk/bin/nearest_neighbors.py

```diff
@@ -0,0 +1,89 @@
+"""
+Retrieve the nearest neighbors of a set of UUIDs.
+"""
+import logging
+import os
+import sys
+
+from smqtk.utils.bin_utils import (
+    basic_cli_parser,
+    utility_main_helper,
+)
+
+from smqtk.algorithms.nn_index import get_nn_index_impls
+from smqtk.utils import plugin
+
+
+__author__ = 'dan.lamanna@kitware.com'
+
+
+def get_cli_parser():
+    parser = basic_cli_parser(__doc__)
+    parser.add_argument('-u', '--uuid-list',
+                        default=None, metavar='PATH',
+                        help='Path to list of UUIDs to calculate nearest '
+                             'neighbors for. If empty, all UUIDs present '
+                             'in the descriptor index will be used.')
+
+    parser.add_argument('-n', '--num',
+                        default=10, metavar='INT', type=int,
```

Comment:
This could have a default of ``None`` to signify getting all neighbors (4. in your list).